core/src/read.jl

```diff
@@ -246,7 +246,102 @@ const conservative_nodetypes = Set{NodeType.T}([
     NodeType.ManningResistance,
 ])
 
-function initialize_allocation!(p::Parameters, config::Config)::Nothing
+function get_source_priority_data(
+    p::Parameters,
+    db::DB,
+    config::Config,
+)::Vector{SOURCE_TUPLE}
+    (; graph) = p
+    (; default_source_priority) = config.allocation
+
+    node_rows = execute(
+        db,
+        "SELECT node_id, node_type, subnetwork_id, source_priority FROM Node ORDER BY subnetwork_id, source_priority",
+    )
+
+    # Build dictionary source type -> default source priority (e.g. "user_demand" => 1000)
+    default_source_priority_dict = Dict{Symbol, Int32}()
+    source_types = propertynames(default_source_priority)
+    for source_type in source_types
+        if source_type == :boundary
+            default_priority = default_source_priority.boundary
+            default_source_priority_dict[:flow_boundary] = default_priority
+            default_source_priority_dict[:level_boundary] = default_priority
+        else
+            default_source_priority_dict[source_type] =
+                getproperty(default_source_priority, source_type)
+        end
+    end
```

Comment:
This seems a bit convoluted, how about?
```julia
default_source_priority_dict = Dict{Symbol, Int32}(key => getfield(s, key) for key in propertynames(s))
default_source_priority_dict[:flow_boundary] = default_source_priority.boundary
default_source_priority_dict[:level_boundary] = default_source_priority.boundary
```
(if needed, followed by a delete of :boundary)